src/topp/FalseDiscoveryRate.cpp

```diff
@@ -171,6 +177,20 @@ class TOPPFalseDiscoveryRate :
       it->assignRanks();
     }
 
+    double psm_fdr = getDoubleOption_("FDR:PSM");
+    if (psm_fdr >= 0)
+    {
+      LOG_INFO << "FDR control: Filtering PSMs..." << endl;
+      IDFilter::filterHitsByScore(pep_ids, psm_fdr / 100.0);
+    }
+
+    double protein_fdr = getDoubleOption_("FDR:protein");
+    if (protein_fdr >= 0)
+    {
+      LOG_INFO << "FDR control: Filtering proteins..." << endl;
```

Comment:
I think correct would be to check if the score type is q-value? FDR filtering should work without recalculation if it already contains a q-value. Will add this.